core/include/sched.h

```diff
@@ -188,6 +188,25 @@ void sched_register_cb(void (*callback)(uint32_t, uint32_t));
 
 #endif
 
+#ifdef BOARD_NATIVE
```

Comment:
You could put it into `cpu.h` then you'll probably won't have to change anything here.